src/server/JasmineGraphInstanceService.cpp

```diff
@@ -2209,7 +2209,164 @@ void *instanceservicesession(void *dummyPt) {
                                                      stoi(serverHostDataPort), filename, filepath);
                 sleep(1);
             }
-        } else if (line.compare(JasmineGraphInstanceProtocol::GRAPH_STREAM_START) == 0) {
+        }
+        else if (line.compare(JasmineGraphInstanceProtocol::COLLECT_BUCKETS_TO_COORDINATOR) == 0) {
+            instance_logger.log("Received : " + JasmineGraphInstanceProtocol::COLLECT_BUCKETS_TO_COORDINATOR, "info");
+            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
+            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+            bzero(data, INSTANCE_DATA_LENGTH);
+
+            //Read coordinator host name
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string orgHostListStr = (data);
+            orgHostListStr = utils.trim_copy(orgHostListStr, " \f\n\r\t\v");
+            instance_logger.log("Received HostName: " + orgHostListStr, "info");
+            vector<string> orgHostList = utils.split(orgHostListStr, '|');
+
+            //Read coordinator port from socket
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string orgPortListStr = (data);
+            orgPortListStr = utils.trim_copy(orgPortListStr, " \f\n\r\t\v");
+            instance_logger.log("Received Port: " + orgPortListStr, "info");
+            vector<string> orgPortList = utils.split(orgPortListStr, '|');
+
+            //Read coordinator data port from socket
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string serverHostDataPort = (data);
+            serverHostDataPort = utils.trim_copy(serverHostDataPort, " \f\n\r\t\v");
+            instance_logger.log("Received Data Port: " + serverHostDataPort, "info");
+
+            //Read graph ID from socket
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            string graphID = (data);
+            graphID = utils.trim_copy(graphID, " \f\n\r\t\v");
+            instance_logger.log("Received Graph ID: " + graphID, "info");
+
+            //Read cluster count from socket
+            bzero(data, INSTANCE_DATA_LENGTH);
+            read(connFd, data, INSTANCE_DATA_LENGTH);
+            int noClusters = stoi(data);
+            graphID = utils.trim_copy(graphID, " \f\n\r\t\v");
+            instance_logger.log("Received Graph ID: " + graphID, "info");
+
+            //Get cluster filenames and send to coordinator
+            int count = 0;
+            int orgCount = orgPortList.size();
+            std::thread *workerThreads = new std::thread[orgCount];
+            for (int i = 0; i < orgCount; i++) {
+                workerThreads[count++] = std::thread(JasmineGraphInstanceService::collectBucketstoCoordinator,
```

Comment:
Change name from `collectBucketstoCoordinator` to `collectBucketsToCoordinator`.

Reply:
Name change done

Reply:
Ok